configure.ac

```diff
@@ -225,12 +226,12 @@ AS_IF([test x"$enable_hardening" != x"no"], [
     )
   ])
 
+  add_hardened_c_flag([-Werror])
   add_hardened_c_flag([-Wall])
   add_hardened_c_flag([-Wextra])
-  add_hardened_c_flag([-Werror])
 
   add_hardened_c_flag([-Wformat])
-  add_hardened_c_flag([-Wformat-security])
+  add_hardened_c_flag([-Wformat -Wformat-security])
```

Comment:
Why this change?

Reply:
> Why this change?

Because, if `gcc -Wformat-security` is run by itself, you get:
```
$ gcc -Wformat-security -o foo foo.c
cc1: warning: ‘-Wformat-security’ ignored without ‘-Wformat’ [-Wformat-security]
```
While it is technically 'recognized', it would error out with -Werror flipped:
```
$ gcc -Werror -Wformat-security -o foo foo.c
cc1: error: ‘-Wformat-security’ ignored without ‘-Wformat’ [-Werror=format-security]
cc1: all warnings being treated as errors
```
So, in my mind, checking for this particular flag's existence on its own is invalid.